### PR TITLE
U/elisabethcvs/fixdisplaynameregex

### LINKED ISSUE #15878: getInitials should strip [xyz] (angled braces) just like it strips (xyz)
### Environment Information

- **Package version(s)**: (fill this out)
- **Browser and OS versions**: (fill this out if relevant)

#### Actual behavior:

RegEx for the code today in
```
const UNWANTED_CHARS_REGEX: RegExp = /\([^)]*\)|[\0-\u001F\!-/:-@\[-`\{-\u00BF\u0250-\u036F\uD800-\uFFFF]/g;

function cleanupDisplayName(displayName: string): string {
  displayName = displayName.replace(UNWANTED_CHARS_REGEX, '');
  displayName = displayName.replace(MULTIPLE_WHITESPACES_REGEX, ' ');
  displayName = displayName.trim();

  return displayName;
}
```

Takes `Kevin Andersen (EEE)` and turns it into `Kevin Andersen` but takes `Kevin Andersen [EEE]` and turns it into `Kevin Andersen EEE` resulting in initials of `KE` in the latter case (undesired).

#### Expected behavior:

Strip the content of [xyz] just like we strip the content of (xyz)
Probably should do the same thing for the curly braces too

### Priorities and help requested:

Are you willing to submit a PR to fix? (Yes, No) Yes

Requested priority: (Blocking, High, Normal, Low) Normal

Products/sites affected: (if applicable)

### PR DESCRIPTION
**Pull request by elisabethcvs**
#### Pull request checklist

- [x] Addresses an existing issue: Fixes #15878 
- [x] Include a change request file using `$ yarn change`

#### Description of changes
 
Currently, getInitials takes Kevin Andersen (EEE) and turns it into Kevin Andersen but takes Kevin Andersen [EEE] and turns it into Kevin Andersen EEE resulting in initials of KE in the latter case (undesired).

I've modified it so that the regex strips [xyz] and {xyz} now as well.

I've also updated:
- the tests for initials, as they previously did not cover these cases
- the Persona initials example to contain the old case and new one
- the persona tests to cover this case as well

I think ideally we would like to backport this to 7, but as it has the potential for breaks, it might be safer just in 8. Open to suggestions here.

### CODE REVIEW

File: packages/utilities/src/initials.ts [hunk 1]
```diff
@@ -3,7 +3,8 @@
  * The first part matches characters within parenthesis, including the parenthesis.
  * The second part matches special ASCII characters except space, plus some unicode special characters.
  */
-const UNWANTED_CHARS_REGEX: RegExp = /\([^)]*\)|[\0-\u001F\!-/:-@\[-`\{-\u00BF\u0250-\u036F\uD800-\uFFFF]/g;
+// eslint-disable-next-line @fluentui/max-len
+const UNWANTED_CHARS_REGEX: RegExp = /[\(\[\{][^\)\]\}]*[\)\]\}]|[\0-\u001F\!-/:-@\[-`\{-\u00BF\u0250-\u036F\uD800-\uFFFF]/g;
```
Comment (mtennoe) [inline packages/utilities/src/initials.ts:7]: Hm, the regex becomes quite unreadable after a while. Can we try to split it into sub-regexes with descriptive names, and then use a combination of different regexes instead? That will make it a lot easier to understand and maintain this in the future

Comment (elisabethcvs) [reply packages/utilities/src/initials.ts:7]: Would splitting it into two for the brackets and then special characters  work, or were you thinking even more broken down then that

Comment (philipzloh) [reply packages/utilities/src/initials.ts]: Is the suggestion to move to something like this?

```
const UNWANTED_CHARS_REGEX: RegExp = /[\(\[\{][^\)\]\}]*[\)\]\}]|[\0-\u001F\!-/:-@\[-`\{-\u00BF\u0250-\u036F\uD800-\uFFFF]/g;

====

var openBrackets = "[\(\[\{][^\)\]\}]";
var notCloseBrackets = "[^\)\]\}]";
var closeBrackets = "[\)\]\}]";
var miscSpecialChars = "[\0-\u001F\!-/:-@\[-`\{-\u00BF\u0250-\u036F\uD800-\uFFFF]";

const UNWANTED_CHARS_REGEX = new RegExp(openBrackets + notCloseBrackets + "*" + closeBrackets + "|" + miscSpecialChars, "g");
```

Comment (philipzloh) [reply packages/utilities/src/initials.ts]: oh Elisabeth's new change is good too

Comment (mtennoe) [reply packages/utilities/src/initials.ts]: Yup, like the latest change! A lot easier to read :)

**Review by mtennoe — APPROVED** [review]

**Pull request merged**